extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampCommentsExtractor.java

```diff
@@ -39,18 +51,77 @@ public InfoItemsPage<CommentsInfoItem> getInitialPage()
 
         final CommentsInfoItemsCollector collector = new CommentsInfoItemsCollector(getServiceId());
 
-        final Elements writings = document.getElementsByClass("writing");
+        final JsonObject collectorsData = JsonUtils.toJsonObject(
+                document.getElementById("collectors-data").attr("data-blob"));
+        final JsonArray reviews = collectorsData.getArray("reviews");
 
-        for (final Element writing : writings) {
-            collector.commit(new BandcampCommentsInfoItemExtractor(writing, getUrl()));
+        for (final Object review : reviews) {
+            collector.commit(
+                    new BandcampCommentsInfoItemExtractor((JsonObject) review, getUrl()));
         }
 
-        return new InfoItemsPage<>(collector, null);
+        if (!collectorsData.getBoolean("more_reviews_available")) {
+            return new InfoItemsPage<>(collector, null);
+        }
+
+        final String trackId = getTrackId();
+        final String token = getNextPageToken(reviews);
+        return new InfoItemsPage<>(collector, new Page(trackId, token));
     }
 
     @Override
     public InfoItemsPage<CommentsInfoItem> getPage(final Page page)
             throws IOException, ExtractionException {
-        return null;
+
+        final CommentsInfoItemsCollector collector = new CommentsInfoItemsCollector(getServiceId());
+
+        final String trackId = page.getUrl();
+        final String token = page.getId();
+        final JsonObject reviewsData = fetchReviewsData(trackId, token);
+        final JsonArray reviews = reviewsData.getArray("results");
+
+        for (final Object review : reviews) {
+            collector.commit(
+                    new BandcampCommentsInfoItemExtractor((JsonObject) review, getUrl()));
+        }
+
+        if (!reviewsData.getBoolean("more_available")) {
+            return new InfoItemsPage<>(collector, null);
+        }
+
+        final String nextToken = getNextPageToken(reviews);
+        return new InfoItemsPage<>(collector, new Page(trackId, nextToken));
```

Comment:
No need to use an intermediate variable here:

```suggestion
        return new InfoItemsPage<>(collector,
                new Page(trackId, getNextPageToken(reviews)));
```